src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SearchIndexEntryJObjectGenerator.cs

```diff
@@ -83,14 +83,31 @@ void ISearchValueVisitor.Visit(DateTimeSearchValue dateTime)
 
         void ISearchValueVisitor.Visit(NumberSearchValue number)
         {
-            AddProperty(SearchValueConstants.NumberName, number.Number);
+            if (number.Low == number.High)
+            {
+                AddProperty(SearchValueConstants.NumberName, number.Low);
```

Comment:
Should we consider writing to low and high here as well? I guess if we are doing migration later, we could just do all of it at once for existing data.

Reply:
Yup, I'll update